### DISCUSSION TITLE
Issues with simulations on Frontera

### DISCUSSION THREAD

**Discussion by vikaskurapati** [SeisSol/SeisSol#1019] [Q&A]
Hi,

I am trying to run SeisSol on Frontera and I am facing some issues. I am using the batch script from https://seissol.readthedocs.io/en/latest/frontera.html. I did not use the spack workflow as there were quite some errors which I could not resolve which it returned.

I installed  all the dependencies manually except eigen which is loaded from the modules available. I receive the following errors:

Rank:        0 | Warning | Ignoring space order from parameter file, using           3
[c204-001:195651:0:195651] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x7fff1442aff8)
[c204-001:195650:0:195650] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x7fff61eb2ff8)
==== backtrace (tid: 195651) ====
 0 0x000000000007d067 __GI__IO_default_xsputn()  :0
 1 0x0000000000048707 _IO_vfprintf_internal()  :0
 2 0x0000000000077e73 _IO_vasprintf()  :0
 3 0x0000000000c24b4d H5E_printf_stack()  ???:0
 4 0x0000000000c16a12 H5P_get()  ???:0

and similar errors and then

[mpiexec@c204-001.frontera.tacc.utexas.edu] HYD_sock_write (../../../../../src/pm/i_hydra/libhydra/sock/hydra_sock_intel.c:360): write error (Bad file descriptor)
[mpiexec@c204-001.frontera.tacc.utexas.edu] HYD_sock_write (../../../../../src/pm/i_hydra/libhydra/sock/hydra_sock_intel.c:360): write error (Bad file descriptor)
[mpiexec@c204-001.frontera.tacc.utexas.edu] HYD_sock_write (../../../../../src/pm/i_hydra/libhydra/sock/hydra_sock_intel.c:360): write error (Bad file descriptor)
[mpiexec@c204-001.frontera.tacc.utexas.edu] HYD_sock_write (../../../../../src/pm/i_hydra/libhydra/sock/hydra_sock_intel.c:360): write error (Bad file descriptor)

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 195650 RUNNING AT c204-001
=   KILLED BY SIGNAL: 11 (Segmentation fault)
===================================================================================

I am unable to understand what the exact error is and how it could be fixed. I reinstalled hdf5 with mpiicc, mpiicpc as it seems that the error lies somewhere related to it.

Thanks in advance!

**Comment by Thomas-Ulrich** [comment]
"I did not use the spack workflow as there were quite some errors"
which error?
I would strongly recommend the spack workflow (and in particular the preinstalled seissol-env module).

**Reply by vikaskurapati** [reply]
seissol-env doesn't exist directly after module use /work2/09160/ulrich/frontera/spack/share/spack/modules/linux-centos7-cascadelake. Mainly due to permissions, I guess. 

When I try to install seissol-env as mentioned on the link, during the step of spack install seissol-env %intel ^intel-mpi, I get the following error:
Error: ProcessError: Command exited with status 77:
    '/tmp/vikaskurapati/spack-stage/spack-stage-gmake-4.4.1-il2n7begrte5dxicsgxjrsnblnbjzndk/spack-src/configure' '--prefix=/home1/09830/vikaskurapati/.spack/bootstrap/store/linux-centos7-x86_64/gcc-8.3.0/gmake-4.4.1-il2n7begrte5dxicsgxjrsnblnbjzndk' '--without-guile' '--disable-nls'

2 errors found in build log:
     6     checking for gawk... gawk
     7     checking whether make sets $(MAKE)... yes
     8     checking whether make supports nested variables... yes
     9     checking whether make supports the include directive... yes (GNU style)
     10    checking for gcc... /home1/09830/vikaskurapati/spack/lib/spack/env/gcc/gcc
     11    checking whether the C compiler works... no
  >> 12    configure: error: in `/tmp/vikaskurapati/spack-stage/spack-stage-gmake-4.4.1-il2n7begrte5dxicsgxjrsnblnbjzndk/spack-src/spack-build':
  >> 13    configure: error: C compiler cannot create executables
     14    See `config.log' for more details

See build log for details:
  /tmp/vikaskurapati/spack-stage/spack-stage-gmake-4.4.1-il2n7begrte5dxicsgxjrsnblnbjzndk/spack-build-out.txt

==> Warning: Skipping build of patchelf-0.18.0-yh5q7zontj5m46ktppomav2uzusidl4t since gmake-4.4.1-il2n7begrte5dxicsgxjrsnblnbjzndk failed
==> Error: patchelf-0.18.0-yh5q7zontj5m46ktppomav2uzusidl4t: Package was not installed
==> Error: cannot bootstrap any of the patchelf executables from spec "patchelf@0.13.1:0.13 %gcc target=x86_64" due to the following failures:
github-actions-v0.5 raised CorruptDatabaseError: error parsing database:
    [Errno 2] No such file or directory: '/home1/09830/vikaskurapati/.spack/cache/indices/164d106a82_d51b60c93a.json'
github-actions-v0.4 raised CorruptDatabaseError: error parsing database:
    [Errno 2] No such file or directory: '/home1/09830/vikaskurapati/.spack/cache/indices/164d106a82_d51b60c93a.json'
spack-install raised AssertionError: 
Run `spack --debug ...` for more detailed errors
